Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -1430,50 +1433,25 @@ void CGameListCtrl::AutomaticColumnWidth()
   }
   else if (GetColumnCount() > 0)
   {
-    int resizable =
-        rc.GetWidth() - (GetColumnWidth(COLUMN_PLATFORM) + GetColumnWidth(COLUMN_BANNER) +
-                         GetColumnWidth(COLUMN_ID) + GetColumnWidth(COLUMN_COUNTRY) +
-                         GetColumnWidth(COLUMN_SIZE) + GetColumnWidth(COLUMN_EMULATION_STATE));
-    if (SConfig::GetInstance().m_showTitleColumn && SConfig::GetInstance().m_showMakerColumn &&
-        SConfig::GetInstance().m_showFileNameColumn)
-    {
-      SetColumnWidth(COLUMN_TITLE, resizable / 3);
-      SetColumnWidth(COLUMN_MAKER, resizable / 3);
-      SetColumnWidth(COLUMN_FILENAME, resizable / 3);
-    }
-    else if (SConfig::GetInstance().m_showMakerColumn &&
-             SConfig::GetInstance().m_showFileNameColumn)
-    {
-      SetColumnWidth(COLUMN_MAKER, resizable / 2);
-      SetColumnWidth(COLUMN_FILENAME, resizable / 2);
-    }
-    else if (SConfig::GetInstance().m_showMakerColumn && SConfig::GetInstance().m_showTitleColumn)
-    {
-      SetColumnWidth(COLUMN_MAKER, resizable / 2);
-      SetColumnWidth(COLUMN_TITLE, resizable / 2);
-    }
-    else if (SConfig::GetInstance().m_showFileNameColumn &&
-             SConfig::GetInstance().m_showTitleColumn)
-    {
-      SetColumnWidth(COLUMN_TITLE, resizable / 2);
-      SetColumnWidth(COLUMN_FILENAME, resizable / 2);
-    }
-    else if (SConfig::GetInstance().m_showMakerColumn)
-    {
-      SetColumnWidth(COLUMN_MAKER, resizable);
-    }
-    else if (SConfig::GetInstance().m_showFileNameColumn)
-    {
-      SetColumnWidth(COLUMN_FILENAME, resizable);
-    }
-    else if (SConfig::GetInstance().m_showTitleColumn)
-    {
-      SetColumnWidth(COLUMN_TITLE, resizable);
-    }
-    else
+    int resizable = rc.GetWidth();
+    std::vector<int> visible_columns;
+
+    for (const auto& c : m_columns)
     {
-      SetColumnWidth(COLUMN_DUMMY, resizable);
+      if (c.visible)
+      {
+        if (c.resizable)
+          visible_columns.push_back(c.id);
+        else
+          resizable -= c.default_width;
+      }
     }
+
+    if (visible_columns.empty())
+      visible_columns.push_back(COLUMN_DUMMY);
+
+    for (const int column : visible_columns)
+      SetColumnWidth(column, resizable / visible_columns.size());
```

Comment:
the expression in the second argument position here needs to be casted to an int. `size()` returns a size_t, so this truncates.